graftm/tree_decorator.py

```diff
@@ -0,0 +1,304 @@
+################################################################################
+################################# - Imports - ##################################
+
+# Local imports
+from graftm.getaxnseq import Getaxnseq
+
+# System imports
+from skbio import TreeNode
+import argparse
+import logging
+import os
+import sys
+
+################################################################################
+################################ - Statics - ###################################
+MAX_RESOLUTION = 7
+
+################################################################################
+################################## - Code - ####################################
+
+class TreeUnrootedException(Exception): 
+    pass
+
+class TreeDecorator:
+    '''Yet another class that decorates trees, only the way I want it.'''
+    
+    
+    def __init__(self, 
+                 tree,
+                 taxonomy,
+                 seqinfo=None):
+        '''
+        Set up empty lists and open and import tree object, and load the
+        taxonomy into a hash
+        '''
+        
+        self.gg_prefixes = ["k__", 'p__', 'c__', 'o__', 'f__', 'g__', 's__']
+        
+        # Set empty list to record nodes that have already been assigned
+        # taxonomy.
+        self.encountered_nodes = {}
+        self.encountered_taxonomies = set()
+        
+        # Read in tree
+        self.tree = tree
+        
+        # Read in taxonomy
+        if seqinfo:
+            gtns = Getaxnseq()
+            logging.info("Importing taxtastic taxonomy from files: %s and %s" % (taxonomy, seqinfo))
+            self.taxonomy =  gtns.read_taxtastic_taxonomy_and_seqinfo(open(taxonomy), open(seqinfo))
+            for id, taxonomy_list in self.taxonomy.iteritems():
+                if len(taxonomy_list) != 7:
```

Comment:
Also, it doe sfind how many ranks are there from the taxonomy. This bit of code is just for filling out the rest of the empty tax strings, so that all tax strings are of a uniform length.

Anyway I replaced the 7 with a constant MAX_RESOLUTION used in the code below